hpedockerplugin/cmd/cmd_deleteshare.py

```diff
@@ -19,93 +23,107 @@ def __init__(self, file_mgr, share_info):
         self._fpg_name = share_info['fpg']
 
     def execute(self):
-        LOG.info("Delting share %s..." % self._share_info['name'])
+        share_name = self._share_info['name']
+        LOG.info("Deleting share %s..." % share_name)
         # Most likely nothing got created at the backend when share is
         # not in AVAILABLE state
-        if self._share_info['status'] != 'AVAILABLE':
-            self._delete_share_from_etcd(self._share_info['name'])
+        if self._share_info['status'] == 'FAILED':
+            LOG.info("Share %s not in FAILED state. Removing from ETCD..."
+                     % share_name)
+            self._delete_share_from_etcd(share_name)
             return json.dumps({u"Err": ''})
+        elif self._share_info['status'] == 'CREATING':
```

Comment:
Before delete API gets called, docker makes inspect call hence share_info will have latest available status information. So it is good to have this check to verify status of share. 
Hence this check looks valid.
@imran-ansari correct me if I am wrong.

Reply:
That's right. Before reaching here, share info is fetched from ETCD and passed here.